### DISCUSSION TITLE
How to Run a Rake Task After Deployment on Web Role Docker Containers in Kamal?

### DISCUSSION THREAD

**Discussion by elassadi** [basecamp/kamal#963] [Q&A]
I’m using Kamal to deploy my Rails application, and I need to run a rake task after each deployment. While I have configured the post-deploy hook, I’ve realized that it runs on the target machine but not within the specific application Docker instances.

I need help figuring out how to ensure the rake task runs inside the Docker container, specifically within the web role.

Any advice on how to target only the web role Docker containers inside the post-deploy hook would be greatly appreciated!

Thanks in advance!

**Comment by nickhammond** [comment]
Hey @elassadi,

It's usually via an entrypoint script, here's the default new Rails app setup that runs migrations:

At the very end of your _Dockerfile_
```
........
# Entrypoint prepares the database.
ENTRYPOINT ["/rails/bin/docker-entrypoint"]

# Start the server by default, this can be overwritten at runtime
EXPOSE 3000
CMD ["./bin/rails", "server"]
```

_bin/docker-entrypoint_
```
#!/bin/bash -e

# If running the rails server then create or migrate existing database
if [ "${*}" == "./bin/rails server" ]; then
  ./bin/rails db:prepare
fi

exec "${@}"
```

More details: https://www.fromthekeyboard.com/how-kamal-runs-your-migrations/
